src/ModelContextProtocol-SemanticKernel/Extensions/KernelExtensions.cs

```diff
@@ -64,6 +64,137 @@ public static async Task<IReadOnlyList<KernelPlugin>> AddToolsFromClaudeDesktopC
+        var registeredPlugins = new List<KernelPlugin>();
+        foreach (var kvp in config.McpSection.McpServers.Where(s => !s.Value.Disabled))
+        {
+            if (kvp.Value.Type == VSCodeServerType.Stdio)
+            {
+                registeredPlugins.Add(await AddMcpFunctionsFromStdioServerAsync(plugins, options =>
+                {
+                    options.Name = kvp.Key;
+                    options.Command = kvp.Value.Command;
+                    options.Arguments = kvp.Value.Args;
+                    options.EnvironmentVariables = kvp.Value.Env;
+                },
+                cancellationToken));
+                continue;
+
+            }
+            // TODO: See if args can be safely extracted from args
+            //else if (kvp.Value.Type == VSCodeServerType.SSE)
+            //{
+
+            //    continue;
+            //}
+            else
+            {
+                throw new NotSupportedException($"Server type {kvp.Value.Type} is not supported yet.");
+            }
+        }
+
+        return registeredPlugins;
+    }
+
+    /// <summary>
+    /// Adds Model Content Protocol plugins from the Visual Studio settings file (<c>settings.json</c>) and adds it into the plugin collection.
+    /// </summary>
+    /// <param name="plugins">The plugin collection to which the new plugin should be added.</param>
+    /// <param name="instanceType">In case of VSCode Insiders, default can be overriden to auto-detect settings location</param>
+    /// <param name="loggerFactory">The optional <see cref="ILoggerFactory"/>.</param>
+    /// <param name="cancellationToken">The optional <see cref="CancellationToken"/>.</param>
+    /// <returns>A list of <see cref="KernelPlugin"/> containing the functions provided in plugins.</returns>
+    public static async Task<IReadOnlyList<KernelPlugin>> AddToolsFromVsCodeConfigAsync(
+        this KernelPluginCollection plugins,
+        string? path = null,
+        VsCodeInstanceType instanceType = VsCodeInstanceType.VSCode,
+        ILoggerFactory? loggerFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        // https://code.visualstudio.com/docs/configure/settings#_user-settingsjson-location
+        var instancePath = instanceType switch
+        {
+            VsCodeInstanceType.VSCode => "Code",
+            VsCodeInstanceType.VSCodeInsiders => "Code - Insiders", // no docs for this, but it's the name on local machine (windows, including white spaces)
+            _ => throw new ArgumentException("Visual Studio Instance Type not supported yet.", nameof(instanceType))
+        };
+
+        var appDataRoaming = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+        var configPath = Path.Combine(appDataRoaming, instancePath, "User", "settings.json");
+        if (!File.Exists(configPath))
+        {
+            // TODO: Add logging that configuration file was not found
```

Comment:
Is logging using ILogger possible here? Or do you want to use console log?
Maybe just return [] silently?